integrations/gcp/gcp_core/gcp_client.py

```diff
@@ -0,0 +1,107 @@
+from typing import Any, AsyncGenerator, Optional
+
+from google.api_core.exceptions import PermissionDenied
+from google.cloud.asset_v1 import (
+    AssetServiceAsyncClient,
+    ContentType,
+    ListFeedsRequest,
+)
+from google.cloud.asset_v1.services.asset_service import pagers
+from loguru import logger
+from port_ocean.context.event import event
+from port_ocean.context.ocean import ocean
+
+from gcp_core.types import CloudAssetInventoryFeed
+from gcp_core.utils import parseProtobufMessages
+
+RESOURCE_METADATA_CONTENT_TYPE = ContentType(1)
+
+
+class GCPClient:
+    def __init__(self, parent: str, service_account_file: str) -> None:
+        self.async_assets_client: AssetServiceAsyncClient = (
+            AssetServiceAsyncClient.from_service_account_file(service_account_file)
+        )
+        self._parent = parent
+
+    @classmethod
+    def create_from_ocean_config(cls) -> "GCPClient":
+        if cache := event.attributes.get("gcp_client"):
+            return cache
+        try:
+            parent = ocean.integration_config["parent"]
+            service_account = ocean.integration_config["service_account_file_location"]
+        except KeyError as e:
+            raise KeyError(f"Missing required integration key: {str(e)}")
+        gcp_client = cls(parent, service_account)
+        event.attributes["gcp_client"] = gcp_client
+        return gcp_client
```

Comment:
I would prefer the gcp client to be ignorant of the existing of integration config or ocean related configuration. 
I think the configuration should be passed to it.